mysql-test/mysql-test-run.pl

```diff
@@ -968,6 +968,11 @@ ($$$)
           mtr_report("***Warnings generated in error logs during shutdown ".
                      "after running tests: $test_list\n\n$report");
           $test_failure= 1;
+          if (scalar @$completed != 0){
+            if ($result == @$completed[-1] && !defined(@$completed[-1]->{warnings})){
+              @$completed[-1]->{warnings}= $report;
+            }
+          }
```

Comment:
this is weird. you attach warnings to the test that weren't necessarily caused by that test. looks wrong.

Reply:
This situation may occur if the warnings were generated, but generated after the tests. These warnings remain only in the log, but not in the XML.